src/jbod/disks.rs

```diff
@@ -116,6 +122,63 @@ pub mod DiskShelf {
 
         fw_revision
     }
+    /// Returns a string with the disk serial number
+    ///
+    /// # Arguments
+    ///
+    /// * `disk` - a string with the device path
+    ///
+    /// # Example
+    /// ```
+    /// let serial = get_disk_serial("/dev/sg100");
+    /// ```
+    ///
+    fn get_disk_serial(disk: String) -> String {
+        let f: &[u8] = &fs::read(disk + "/vpd_pg80").unwrap();
```

Comment:
I would suggest replace the `let f: &[u8] = &fs::read(disk + "/vpd_pg80").unwrap();` with the following approach:

```
let f: Vec<u8> = if let Ok(t) = fs::read(disk + "/vpd_pg80") {
     t
} else {
     let t = "N/A";
     t.as_bytes().to_vec()
};
```

Reply:
What about doing the following ?

```rust
let f: Vec<u8> = fs::read(disk + "/vpd_pg80").expect("N/A");
```

Reply:
That would do the trick too!

Reply:
Sorry, I made couple more tests and the .expect will also panic if the file doesn't exist. Example:

```
use std::fs;
fn main() {
    let f = fs::read("/tmp/a").expect("N/A");
    println!("f: {:?}", f);
}
```
I think that solution I proposed might be the safest one.

Reply:
A little late but I think you guys were looking for [unwrap_or](https://doc.rust-lang.org/std/result/enum.Result.html#method.unwrap_or)

Here it would have been somthing like
```rust
let f: Vec<u8> =  fs::read(disk + "/vpd_pg80").unwrap_or( b"N/A".to_vec())
```

(The error and option apis provide quite a lot of methods to go out or transform that struct)